packages/components/tag/src/Tag.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import type { TagType, TagProps } from './types';
+import { Container } from './Tag.styled';
+
+const statusTagTypeMap: { [key: string]: TagType } = {
+  published: 'positive',
+  draft: 'warning',
+  archived: 'negative',
+  changed: 'primary',
+  deleted: 'negative',
+  new: 'primary-filled',
+};
+
+function Tag(props: TagProps, ref: React.Ref<HTMLDivElement>) {
+  const {
+    className,
+    children,
+    tagType = 'primary' as TagType,
+    entityStatusType,
+    testId = 'cf-ui-tag',
+    ...otherProps
+  } = props;
+
+  const calculatedTagType = entityStatusType
+    ? statusTagTypeMap[entityStatusType]
+    : tagType;
+
+  return (
+    <Container
```

Comment:
💭 how would I expose the html element for `as` prop with styled `Container`?

Reply:
Yeah, we can do this if we want to for a11y purposes, but I'm don't have an opinion on this.